### ISSUE TITLE
Social media icons with 6.4.15.2 security update

### ISSUE THREAD

**Issue by LiaraAlis** [digitalmanufaktur/shopware6-manu-theme#20]
It is needed to change the `icon.html.twig` template according to this changes in the shopware core:
https://github.com/shopware/platform/blob/6.4.15.2/UPGRADE-6.4.md#64152

Otherwise, social media icons are not displayed. This also happens when the security plugin is installed and active.

**Comment by LiaraAlis** [comment]
I have made the changes locally, but this does not solve the problem.
Here there is probably additionally also a Shopware Bug: https://issues.shopware.com/issues/NEXT-23562

(Please also note the last comments from Tobias Berge in the Shopware issue tracker)

**Comment by lutek** [comment]
you need to change theme.json asset path from src to dist, and icon.html.twig in my case i have manu theme as parent theme.

```
    "asset": [
        "@Storefront",
        "@DmfManuTheme",
        "app/storefront/dist/assets"
    ],
```

```
{% sw_extends '@Storefront/storefront/utilities/icon.html.twig' %}

{% block utilities_icon %}
    {% if pack is not defined %}
        {% set pack = 'default' %}
    {% endif %}
    {% set customIcon = source('@FwTheme/assets/icon/'~ pack ~'/'~ name ~'.svg', ignore_missing = true) %}
    {% if (customIcon) %}
        {% set styles = [ size, color, rotation, flip, class ] %}
        <span
            class="icon icon-{{ name }}{% for entry in styles %}{% if entry != "" %} icon-{{ entry }}{% endif %}{% endfor %}">
            {{ customIcon|sw_icon_cache|raw }}
        </span>
    {% else %}
        {{ parent() }}
    {% endif %}
{% endblock %}
```

**Comment by LiaraAlis** [comment]
@lutek Yes, but it is not fixed in the manu theme and using manu theme without a child theme results in the problem described in this issue.
And icons that came from the manu theme are also not loaded with your change.

**Comment by OlliZabal** [comment]
> you need to change theme.json asset path from src to dist, and icon.html.twig in my case i have manu theme as parent theme.
> 
> ```
>     "asset": [
>         "@Storefront",
>         "@DmfManuTheme",
>         "app/storefront/dist/assets"
>     ],
> ```
> 
> ```
> {% sw_extends '@Storefront/storefront/utilities/icon.html.twig' %}
> 
> {% block utilities_icon %}
>     {% if pack is not defined %}
>         {% set pack = 'default' %}
>     {% endif %}
>     {% set customIcon = source('@FwTheme/assets/icon/'~ pack ~'/'~ name ~'.svg', ignore_missing = true) %}
>     {% if (customIcon) %}
>         {% set styles = [ size, color, rotation, flip, class ] %}
>         <span
>             class="icon icon-{{ name }}{% for entry in styles %}{% if entry != "" %} icon-{{ entry }}{% endif %}{% endfor %}">
>             {{ customIcon|sw_icon_cache|raw }}
>         </span>
>     {% else %}
>         {{ parent() }}
>     {% endif %}
> {% endblock %}
> ```

Thanks @lutek for this! As @LiaraAlis mentioned, this does not have an affect yet.
I went deeper and I got it to work based on your changes @lutek I changed the file "icon.html.twig" as following:

```
{% sw_extends '@Storefront/storefront/utilities/icon.html.twig' %}

{% block utilities_icon %}
    {% if pack is not defined %}
        {% set pack = 'default' %}
    {% endif %}
    {% if namespace is not defined %}
        {% set namespace = 'DmfManuTheme' %}
    {% endif %}
    {% set customIcon = source('@' ~ themeIconConfig[pack].namespace ~ '/' ~ themeIconConfig[pack].path ~'/'~ name ~'.svg', ignore_missing = true) %}
    {% if (customIcon) %}
        {% set styles = [ size, color, rotation, flip, class ] %}
        <span
            class="icon icon-{{ name }}{% for entry in styles %}{% if entry != "" %} icon-{{ entry }}{% endif %}{% endfor %}">
            {{ customIcon|raw }}
        </span>
    {% else %}
        {{ parent() }}
    {% endif %}
{% endblock %}
```

Of course, the icons need to be in DmfManuTheme\src\Resources\app\storefront\dist\assets\icon\default

**Comment by annakoehler** [comment]
Thank you for your comments!
We published v2.0.6 in which this problem in DmfManuTheme is solved.
